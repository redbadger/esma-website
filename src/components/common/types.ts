export type Fluid = {
  originalName: string;
  base64: string;
  sizes: string;
  src: string;
  srcSet: string;
  aspectRatio: number;
};

export type ChildImageSharp = {
  fluid: Fluid;
};

export type Node = {
  childImageSharp: ChildImageSharp;
};

export type ImageNode = {
  node: Node;
};
