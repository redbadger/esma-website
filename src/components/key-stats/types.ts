import { Stat as PieStat } from "../pie";

export enum KeyStatsCategories {
  Individual = "individual",
  Society = "society",
  Businesses = "businesses",
}

export type ArticleProps = {
  statClasses: (x: string) => string;
};

/// PieProps
/// position: x, y,
/// denotes where on our 300 * 591 map of the UK to put the pie chart
export type PieMapOverlayProps = {
  position: { x: number; y: number };
  pieStats: PieStat[];
  name: string;
};

export type LegendProps = {
  label: string;
  colour: string;
}[];
