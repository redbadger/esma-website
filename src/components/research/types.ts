export type PillData = {
  name: string;
  href: string;
};

export type PillsProps = {
  pills: Array<PillData>;
  colorActive: string;
  currentPillIndex: number;
};

export type ResearchPageMetaData = {
  order: number;
  parent: string;
  title: string;
  color: string;
  bgImageName: string;
  footnotes: FootNoteData[];
};

export type FootNoteData = {
  id: string;
  text: string;
  destination: string;
};

