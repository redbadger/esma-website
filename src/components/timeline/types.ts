export type TimelineEntryData = {
  title: string;
  imageName: string;
  description: string;
  destination: string;
  icon: React.ElementType;
  anchor: string;
};

export type TimelineLegendData = {
  title: string;
  icon: React.ElementType;
  color: string;
  anchor: string;
};
