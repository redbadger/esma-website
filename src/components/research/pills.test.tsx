import React from "react";
import renderer from "react-test-renderer";
import Pills from "./pills";
import { PillData } from "./types";

describe("Pills", () => {
  const samplePills: PillData[] = [
    { name: "Executive summary", href: "/" },
    { name: "Early Years Foundation Stage", href: "/" },
    { name: "Family Circumstances", href: "/" },
    { name: "Social class and underachievement", href: "/" },
    { name: "School cultures", href: "/" },
    { name: "Geographical educational inequality", href: "/" },
    { name: "Type and quality of schooling", href: "/" },
    { name: "Low aspirations", href: "/" },
    { name: "Access to career advice", href: "/" },
    { name: "Gaps in attainment", href: "/" },
    { name: "Impact of Covid-19", href: "/" },
  ];

  const colorActive = "yellow";

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Pills
          pills={samplePills}
          colorActive={colorActive}
          currentPillIndex={0}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
