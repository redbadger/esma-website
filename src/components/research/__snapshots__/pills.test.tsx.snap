// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pills renders correctly 1`] = `
<nav
  className="css-15dfcdf-dropdownCssMobile-cssListItem-cssList-cssPills-highlightColorCss-Pills"
>
  <div
    className="dropdown"
    onClick={[Function]}
  >
    Contents
    <span
      className="icon "
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-up fa-w-14 "
        data-icon="chevron-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <ol
    className="pill-list"
  >
    <li>
      <a
        className="current-page"
        href="/"
      >
        Executive Summary
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Early Years Foundation Stage
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Family Circumstances
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Social class and underachievement
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        School cultures
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Geographical educational inequality
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Type and quality of schooling
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Low aspirations
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Access to career advice
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Gaps in attainnment
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/"
      >
        Impact of COVID-19
      </a>
    </li>
  </ol>
</nav>
`;
